### PR TITLE
Support overriding the node environment in dev and test

### LINKED ISSUE #60: Support overriding the node environment in dev and test
It is a much better experience to have NODE_ENV overridden 
when running `fusion dev` and `fusion test` than throwing.
In production, we should still error on the side of caution
and throw, however in dev/test it should suffice to log a warning

### PR DESCRIPTION
**Pull request by ganemone**
Fixes #60

### CODE REVIEW

File: test/cli/dev.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ test('`fusion dev` works', async t => {
   const entry = path.resolve(dir, entryPath);
 
   const {stop} = await run(
-    `dev --dir=${dir} --no-open=true --port=${await getPort()}`
+    `dev --dir=${dir} --no-open --port=${await getPort()}`
```
Comment (davidkearns) [inline test/cli/dev.js]: Unrelated change?

File: build/compiler.js [hunk 2]
```diff
@@ -105,6 +105,19 @@ function getConfig({target, env, dir, watch, cover}) {
   // 'test' and 'production' entries should both map to NODE_ENV='production'
   const nodeEnv = env === 'development' ? 'development' : 'production';
 
+  // Allow overrides with a warning for `dev` and `test` commands. In production builds, throw if NODE_ENV is not `production`.
+  const nodeEnvBanner =
+    `if(process.env.NODE_ENV && process.env.NODE_ENV !== '${nodeEnv}') {` +
+    `if ('${env}' === 'production') {` +
```
Comment (lhorie) [inline build/compiler.js]: Better to do this check at compile time rather than bundling the checks into the runtime

Comment (davidkearns) [inline build/compiler.js]: The logic here is sound, but I wonder if there's any (minuscule) benefit of pulling this particular `if/else` out of the template since we can resolve these values earlier at build time.

**Comment by ganemone** [comment]
!merge
